### DISCUSSION TITLE
[Feature Request] Drag & drop with snap to layout

### DISCUSSION THREAD

**Discussion by 3hhh** [qtile/qtile#5073] [Ideas] [answered]
I managed to reproduce 95% of my awesomeWM config in qtile, which is awesome! Thanks a lot to the devs!

One of the few things I didn't manage to reproduce:
In awesomeWM it's possible to mod+Click a window, drag it somewhere and it'll snap to the layout in the position where it was dropped. This is useful to quickly shuffle windows around without floating.

I'll leave it here as a potential idea for the future.

**Comment by elParaguayo** [comment, marked as answer]
Some layouts can do this already.

You need to change the function that's called on the drag from `set_position_floating` to `set_position`.

**Reply by 3hhh** [reply]
Thanks for the hint! I'll see whether I can get it working with MonadTall...

**Reply by 3hhh** [reply]
I can confirm it works, thanks again! :-)
